cassandra/cluster.py

```diff
@@ -2189,7 +2189,7 @@ def shutdown(self):
             else:
                 self.is_shutdown = True
 
-        for pool in self._pools.values():
+        for pool in list(self._pools.values()):
```

Comment:
looks good if this fixes the issue. Note that six.itervalues would have probably worked here as well. It might be harder to understand why we need a list() here in the future. The decision is your.